File: src/main/java/com/pocketempire/config/FactionConfigLoader.java
package com.pocketempire.config;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.InputStream;
import java.util.Collection;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class FactionConfigLoader {
    private static final Map<String, FactionConfig> configs;

    static {
        configs = loadConfigs();
    }

    private static Map<String, FactionConfig> loadConfigs() {
        try {
            ObjectMapper mapper = new ObjectMapper();
            InputStream is = FactionConfigLoader.class.getResourceAsStream("/factions.json");
            FactionConfig[] configs = mapper.readValue(is, FactionConfig[].class);

            Map<String, FactionConfig> map = new HashMap<>();
            for (FactionConfig config : configs) {
                map.put(config.getId(), config);
            }
            return map;
        } catch (Exception e) {
            e.printStackTrace();
            return Collections.emptyMap();
        }
    }

    public static FactionConfig getConfig(String id) {
        return configs.get(id);
    }

    public static Collection<FactionConfig> getAll() {
        return configs.values();
    }
}

File: src/main/kotlin/com/pocketempire/config/FactionConfigLoader.kt
package com.pocketempire.config

import com.fasterxml.jackson.databind.ObjectMapper

object FactionConfigLoader {

    private val configs: Map<String, FactionConfig> = loadConfigs()

    private fun loadConfigs(): Map<String, FactionConfig> {
        return try {
            val inputStream = requireNotNull(
                FactionConfigLoader::class.java
                    .getResourceAsStream("/factions.json")
            ) {
                "Could not find /factions.json"
            }

            inputStream.use {
                val factions = ObjectMapper().readValue(
                    it,
                    Array<FactionConfig>::class.java
                )

                factions.associateBy { faction -> faction.id }
            }
        } catch (e: Exception) {
            e.printStackTrace()
            emptyMap()
        }
    }

    @JvmStatic
    fun getConfig(id: String): FactionConfig? =
        configs[id]

    @JvmStatic
    fun getAll(): Collection<FactionConfig> =
        configs.values
}